Common/include/geometry_structure.hpp

```diff
@@ -1912,6 +1912,35 @@ class CPhysicalGeometry : public CGeometry {
    */
   void PrepareCGNSAdjacency(CConfig *config);
   
+  /*!
+   * \brief Routine to check the type of a CGNS element and to obtain other information about its containing section.
+   * \param[in] config        - definition of the particular problem.
+   * \param[in] val_elem_type - current CGNS element type (input).
+   * \param[in] val_vtk_type  - VTK index identifier (output).
+   * \return String name of the current element type.
+   */
+#ifdef HAVE_CGNS
+  string GetCGNSElementType(CConfig     *config,
+                          ElementType_t val_elem_type,
+                          int           *val_vtk_type);
+#endif
+  
+  /*!
+   * \brief Routine to check whether a CGNS section is an interior or boundary section.
+   * \param[in] config        - definition of the particular problem.
+   * \param[in] val_fn        - CGNS file identifier.
+   * \param[in] val_base      - current CGNS database index.
+   * \param[in] val_zone      - current CGNS zone index.
+   * \param[in] val_section   - current CGNS section index.
+   * \param[in] val_interior  - boolean flag for whether this is an interior or boundary section to be returned.
```

Comment:
I think you mean [out] here, same for val_vtk_type in GetCGNSElementType.
Also I think passing these outputs by reference would be safer and more expressive, looking at just the function signature someone could think this is an array.

Reply:
Yep, will address this. I'm still not perfectly happy with the entire implementation, so a little more refactoring is on the way..